common/authorization/oauthAutorizer_test.go

```diff
@@ -0,0 +1,218 @@
+	tokenExpiredIat := `eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwib
+		mFtZSI6IkpvaG4gRG9lIiwicGVybWlzc2lvbiI6InJlYWQiLCJkb21haW4iOiJ0ZXN0LWRvbWFpbiIsImlhdCI6MTY
+		yNjMzNjQ2MywiVFRMIjoxfQ.P_T3O54F_aiHcaMwyeh2GXtzgWhyKSLkuu8rtGAylK0HOsHYRIkbjdx251kaDEf2B-
+		QP6KKCiXhDgZ_Q42Tb477zjl9IYGRqEj9JZ7PwGuRWCEZWUaFHgB4XmkviHDMamBB5jqg2I2XYklyNO3r2m45_AcQ3
+		dAU4uLiwBwSVKy_YsMldEvGKMC86JvGcYPhu-LLvrJSViQVyuBGjUor6YREuadAZHyKuoMunLq5b_BW2hTf_67kGiy
+		RL5_DxBBGbiNeHDPNoBUNUAx4Nbe1rAckREL8VULVFC_HZ0bDiM7KMJJ0t6zLcgP8Z3Q3341nfhv9r3qG_6U343ZgT
+		PZfQNQ`
+
+	re := regexp.MustCompile(`\r?\n?\t`)
+	token = re.ReplaceAllString(token, "")
+	tokenExpiredIat = re.ReplaceAllString(tokenExpiredIat, "")
+
+	ctx := context.Background()
+	ctx, call := encoding.NewInboundCall(ctx)
+	err := call.ReadFromRequest(&transport.Request{
+		Headers: transport.NewHeaders().With(common.AuthorizationTokenHeaderName, token),
+	})
+	assert.NoError(t, err)
+
+	att := Attributes{
+		Actor:      "John Doe",
+		APIName:    "",
+		DomainName: "test-domain",
+		TaskList:   nil,
+		Permission: PermissionRead,
+	}
+
+	return Mocks{
+		cfg:             cfg,
+		ctx:             ctx,
+		token:           token,
+		tokenExpiredIat: tokenExpiredIat,
+		att:             att,
+	}
+}
+
+func TestCorrectPayload(t *testing.T) {
+	mocks := getMocksBase(t)
+	authorizer := NewOAuthAuthorizer(mocks.cfg)
+	result, err := authorizer.Authorize(mocks.ctx, &mocks.att)
+	assert.NoError(t, err)
+	assert.Equal(t, result.Decision, DecisionAllow)
+}
+
+func TestIncorrectPublicKey(t *testing.T) {
+	mocks := getMocksBase(t)
+	mocks.cfg.JwtCredentials.PublicKey = "incorrectPublicKey"
+	authorizer := NewOAuthAuthorizer(mocks.cfg)
+	result, err := authorizer.Authorize(mocks.ctx, &mocks.att)
+	assert.EqualError(t, err, "failed to parse PEM block containing the public key")
+	assert.Equal(t, result.Decision, DecisionDeny)
+}
+
+func TestIncorrectAlgorithm(t *testing.T) {
+	mocks := getMocksBase(t)
+	mocks.cfg.JwtCredentials.Algorithm = "SHA256"
+	authorizer := NewOAuthAuthorizer(mocks.cfg)
+	result, err := authorizer.Authorize(mocks.ctx, &mocks.att)
+	assert.EqualError(t, err, "jwt: algorithm is not supported")
+	assert.Equal(t, result.Decision, DecisionDeny)
+}
+
+func TestMaxTTLLargerInToken(t *testing.T) {
+	mocks := getMocksBase(t)
+	mocks.cfg.MaxJwtTTL = 1
+	authorizer := NewOAuthAuthorizer(mocks.cfg)
+	result, _ := authorizer.Authorize(mocks.ctx, &mocks.att)
+	//assert.EqualError(t, err, "TTL in token is larger than MaxTTL allowed")
```

Comment:
I was evaluating the error before but now that the error is not returning anything idk how to check the right message. Do you know if you have somewhere in the code a listener/mock for the log? @longquanzheng

Reply:
We don’t have a conversation to verify logging. If you want you can use mockgen to create the mocking instance for test.